db/multi_cf_iterator.h

```diff
@@ -72,18 +73,65 @@ class MultiCfIterator : public Iterator {
     const Comparator* comparator_;
   };
 
+  class MultiCfMaxHeapItemComparator {
+   public:
+    explicit MultiCfMaxHeapItemComparator(const Comparator* comparator)
+        : comparator_(comparator) {}
+
+    bool operator()(const MultiCfIteratorInfo& a,
+                    const MultiCfIteratorInfo& b) const {
+      assert(a.iterator);
+      assert(b.iterator);
+      assert(a.iterator->Valid());
+      assert(b.iterator->Valid());
+      int c = comparator_->Compare(a.iterator->key(), b.iterator->key());
+      assert(c != 0 || a.order != b.order);
+      return c == 0 ? a.order - b.order > 0 : c < 0;
+    }
+
+   private:
+    const Comparator* comparator_;
+  };
+
   const Comparator* comparator_;
   using MultiCfMinHeap =
       BinaryHeap<MultiCfIteratorInfo, MultiCfMinHeapItemComparator>;
+  using MultiCfMaxHeap =
+      BinaryHeap<MultiCfIteratorInfo, MultiCfMaxHeapItemComparator>;
   MultiCfMinHeap min_heap_;
-  // TODO: MaxHeap for Reverse Iteration
+  MultiCfMaxHeap max_heap_;
```

Comment:
FYI, this is a potential long-term efficiency concern. Existing merging iterator uses a unique_ptr for the reversing heap to minimize ctor/dtor costs in the normal case. More alternatives are proposed here: https://github.com/facebook/rocksdb/pull/9035

Related: Looks like the `MultiCfIterator` has three copies of `Comparator*` in its representation.